airflow/providers/docker/operators/docker.py

```diff
@@ -304,21 +304,24 @@ def _run_image_with_mounts(self, target_mounts, add_tmp_variable: bool) -> Optio
             working_dir=self.working_dir,
             tty=self.tty,
         )
-        lines = self.cli.attach(container=self.container['Id'], stdout=True, stderr=True, stream=True)
+        logstream = self.cli.attach(container=self.container['Id'], stdout=True, stderr=True, stream=True)
         try:
             self.cli.start(self.container['Id'])
 
-            line = ''
+            log_chunk = ''
             res_lines = []
             return_value = None
-            for line in lines:
-                if hasattr(line, 'decode'):
-                    # Note that lines returned can also be byte sequences so we have to handle decode here
-                    line = line.decode('utf-8')
-                line = line.strip()
-                res_lines.append(line)
-                self.log.info(line)
+            for log_chunk in logstream:
+                if hasattr(log_chunk, 'decode'):
+                    # Note that log_chunk returned can also be byte sequences so we have to handle decode here
+                    log_chunk = log_chunk.decode('utf-8')
+                log_chunk = log_chunk.strip()
+                res_lines.append(log_chunk)
+                self.log.info(log_chunk)
             result = self.cli.wait(self.container['Id'])
+            # after container has exited, grab the entire log ignoring the chunked log stream that was used with attach
+            # self.cli.logs uses docker's /containers/{id}/logs, while self.cli.attach uses /containers/{id}/attach
+            lines = self.cli.logs(container=self.container['Id'], stdout=True, stderr=True, stream=True)
```

Comment:
@uranusjr  - I'm going to finish this one up.  Is there a reason we're pulling the logs again instead of reusing the parsed logs from above?

Reply:
`logstream` only contains logs starting from the attach happens and may lose things between it and container startup. I’m not sure if it makes a difference, but honestly the additional API call is pretty cheap anyway, and might actually be more efficient with `xcom_all = False` because we don’t need to put all the previous logs into memory.

Reply:
Sure, but we're calling `attach` before we call `cli.start()` so that should mean the stream has everything from the beginning.  I'm happy to go either way, just thought it might be simplified while we were in there.

Reply:
According to #9164, attching too slow and missing a line is a real problem though? I feel it’s just much safer to have a separate API call.